### DISCUSSION TITLE
[Metrics] Histogram and summary differences in opentelemetry

### DISCUSSION THREAD

**Discussion by zmapleshine** [open-telemetry/opentelemetry-java#3767] [Q&A]
Hi all, I find a strange phenomenon that, in my project, metrics exporter was configured as "Prometheus"  ,unlike Micrometer, the indicator of summary generated in otel is output based on a specific time window. Where can the value of this time window be set? 

The metrics names of histogram type are the same as those of the summary type. They all have the suffix "_sum,_count", but the values of the histogram type are cumulative values.

This makes it difficult to configure the chart if it is not consistent. For example, for **Summary** type metrics,the computationally time-consuming "promql" is:

```text
sum(http_client_duration_sum) by (net_peer_name)
/
sum(http_client_duration_count) by (net_peer_name)
```

But once this metric is switched to a histogram representation, the "promql" statement should be as :
```text
sum(rate(http_client_duration_sum[$interval])) by (net_peer_name)
/
sum(rate(http_client_duration_count[$interval])) by (net_peer_name)

```

**Comment by jkwatson** [comment]
@zmapleshine what version of our libraries are you using? @jsuereth has been reworking a lot of our metrics code, so knowing what version you're on could be useful.

**Reply by zmapleshine** [reply]
> @zmapleshine what version of our libraries are you using? @jsuereth has been reworking a lot of our metrics code, so knowing what version you're on could be useful.

Hey @jkwatson ,the SDK version I am using is 1.6.0.
I haven't tried a newer version than this.
I have seen the latest（1.7.x）release change log and there are no changes related to the summary or histogram.

**Reply by jkwatson** [reply]
@jsuereth can confirm, but I believe that the generation of a summary is going away altogether, in favor of histograms, so this may be moot.

**Reply by jsuereth** [reply]
Yes, summary is not included in 1.7.x, so things will be changing here.

Regarding your question on how to query, you should be using `histogram_quantile()`, e.g.


```
histogram_quantile(0.95, sum(rate(http_client_duration[5m])) by (le))
```

**Reply by zmapleshine** [reply]
@jsuereth @jkwatson Oh.. I'm sorry if my description is not clear. I use the SDK to output the client metrics for components such as JDBC, Redis, etc., and can use custom switches to control whether the histogram is turned on or not. Not all histograms can be turned on because of the size of the cardinality. The ``promql`` for **calculating the average elapsed time per unit of time** is the same whether the summary or the histogram is used, except for the quantile line.
